Dependency `yappi==1.3.3` is failed to install on Apple M1

macOS: 13.0.1 (22A400)
chip: Apple M1

```python
➜ pip install yappi==1.3.3

Collecting yappi==1.3.3
  Using cached yappi-1.3.3.tar.gz (58 kB)
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: yappi
  Building wheel for yappi (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-10.9-universal2-cpython-39
      copying yappi/yappi.py -> build/lib.macosx-10.9-universal2-cpython-39
      running build_ext
      building '_yappi' extension
      creating build/temp.macosx-10.9-universal2-cpython-39
      creating build/temp.macosx-10.9-universal2-cpython-39/yappi
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -iwithsysroot/System/Library/Frameworks/System.framework/PrivateHeaders -iwithsysroot/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Headers -arch arm64 -arch x86_64 -Werror=implicit-function-declaration -I/Users/<user>/Projects/github.com/Tribler/tribler/venv/include -I/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Headers -c yappi/_yappi.c -o build/temp.macosx-10.9-universal2-cpython-39/yappi/_yappi.o
      In file included from yappi/_yappi.c:10:
      yappi/config.h:4:10: fatal error: 'Python.h' file not found
      #include "Python.h"
               ^~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for yappi
  Running setup.py clean for yappi
Failed to build yappi
Installing collected packages: yappi
  Running setup.py install for yappi ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for yappi did not run successfully.
  │ exit code: 1
  ╰─> [19 lines of output]
      running install
      /Users/<user>/Projects/github.com/Tribler/tribler/venv/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build/lib.macosx-10.9-universal2-cpython-39
      copying yappi/yappi.py -> build/lib.macosx-10.9-universal2-cpython-39
      running build_ext
      building '_yappi' extension
      creating build/temp.macosx-10.9-universal2-cpython-39
      creating build/temp.macosx-10.9-universal2-cpython-39/yappi
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -iwithsysroot/System/Library/Frameworks/System.framework/PrivateHeaders -iwithsysroot/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Headers -arch arm64 -arch x86_64 -Werror=implicit-function-declaration -I/Users/<user>/Projects/github.com/Tribler/tribler/venv/include -I/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Headers -c yappi/_yappi.c -o build/temp.macosx-10.9-universal2-cpython-39/yappi/_yappi.o
      In file included from yappi/_yappi.c:10:
      yappi/config.h:4:10: fatal error: 'Python.h' file not found
      #include "Python.h"
               ^~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> yappi

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

https://github.com/sumerc/yappi/issues/79#issuecomment-920615470

--- a/requirements-core.txt
+++ b/requirements-core.txt
@@ -19,7 +19,7 @@ PyOpenSSL==21.0.0
 pyyaml==6.0
 sentry-sdk==1.5.0
 service-identity==21.1.0
-yappi==1.3.3
+yappi==1.4.0
 yarl==1.7.2 # keep this dependency higher than 1.6.3. See: https://github.com/aio-libs/yarl/issues/517
 bitarray==2.5.1
 pyipv8==2.8.0

